CI: Azure Main coverage job failure

Showing up in recent PRs:
- gh-14798
- gh-14801

Traceback below (the SciPy build seems to succeed, followed by very rapid failure of the test suite). 

```
2021-10-04T02:17:11.8302221Z ##[section]Starting: Run tests
2021-10-04T02:17:11.8308037Z ==============================================================================
2021-10-04T02:17:11.8308507Z Task         : Command line
2021-10-04T02:17:11.8308897Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2021-10-04T02:17:11.8309287Z Version      : 2.182.0
2021-10-04T02:17:11.8309579Z Author       : Microsoft Corporation
2021-10-04T02:17:11.8309969Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2021-10-04T02:17:11.8310429Z ==============================================================================
2021-10-04T02:17:11.9506396Z Generating script.
2021-10-04T02:17:11.9508988Z ========================== Starting Command Output ===========================
2021-10-04T02:17:11.9533298Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/c83ea0f6-e83b-41a6-9550-920919f9ee94.sh
2021-10-04T02:17:12.0322585Z Removing previous GCOV .gcda files...
2021-10-04T02:17:12.0516894Z Building, see build.log...
2021-10-04T02:17:16.5589029Z Build OK (0:00:04.506804 elapsed)
2021-10-04T02:17:17.0718687Z Traceback (most recent call last):
2021-10-04T02:17:17.0720170Z   File "runtests.py", line 565, in <module>
2021-10-04T02:17:17.0720891Z     main(argv=sys.argv[1:])
2021-10-04T02:17:17.0721325Z   File "runtests.py", line 316, in main
2021-10-04T02:17:17.0725454Z     result = test(args.mode,
2021-10-04T02:17:17.0727214Z   File "/home/vsts/work/1/s/build/testenv/lib/python3.8/site-packages/scipy/_lib/_testutils.py", line 67, in __call__
2021-10-04T02:17:17.0732216Z     code = pytest.main(pytest_args)
2021-10-04T02:17:17.0733722Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/config/__init__.py", line 143, in main
2021-10-04T02:17:17.0735601Z     config = _prepareconfig(args, plugins)
2021-10-04T02:17:17.0762454Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/config/__init__.py", line 318, in _prepareconfig
2021-10-04T02:17:17.0763580Z     config = pluginmanager.hook.pytest_cmdline_parse(
2021-10-04T02:17:17.0764615Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
2021-10-04T02:17:17.0765574Z     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
2021-10-04T02:17:17.0766618Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
2021-10-04T02:17:17.0769929Z     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
2021-10-04T02:17:17.0770763Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 55, in _multicall
2021-10-04T02:17:17.0771228Z     gen.send(outcome)
2021-10-04T02:17:17.0771890Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/helpconfig.py", line 100, in pytest_cmdline_parse
2021-10-04T02:17:17.0772395Z     config: Config = outcome.get_result()
2021-10-04T02:17:17.0773076Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_result.py", line 60, in get_result
2021-10-04T02:17:17.0773532Z     raise ex[1].with_traceback(ex[2])
2021-10-04T02:17:17.0774196Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 39, in _multicall
2021-10-04T02:17:17.0774664Z     res = hook_impl.function(*args)
2021-10-04T02:17:17.0775356Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1003, in pytest_cmdline_parse
2021-10-04T02:17:17.0775840Z     self.parse(args)
2021-10-04T02:17:17.0776475Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1283, in parse
2021-10-04T02:17:17.0782683Z     self._preparse(args, addopts=addopts)
2021-10-04T02:17:17.0784178Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1191, in _preparse
2021-10-04T02:17:17.0789068Z     self.hook.pytest_load_initial_conftests(
2021-10-04T02:17:17.0790072Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_hooks.py", line 265, in __call__
2021-10-04T02:17:17.0790705Z     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
2021-10-04T02:17:17.0791564Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_manager.py", line 80, in _hookexec
2021-10-04T02:17:17.0792184Z     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
2021-10-04T02:17:17.0792984Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 60, in _multicall
2021-10-04T02:17:17.0793537Z     return outcome.get_result()
2021-10-04T02:17:17.0794260Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_result.py", line 60, in get_result
2021-10-04T02:17:17.0794820Z     raise ex[1].with_traceback(ex[2])
2021-10-04T02:17:17.0795574Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pluggy/_callers.py", line 39, in _multicall
2021-10-04T02:17:17.0796116Z     res = hook_impl.function(*args)
2021-10-04T02:17:17.0796918Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pytest_cov/plugin.py", line 149, in pytest_load_initial_conftests
2021-10-04T02:17:17.0797547Z     plugin = CovPlugin(options, early_config.pluginmanager)
2021-10-04T02:17:17.0798321Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pytest_cov/plugin.py", line 198, in __init__
2021-10-04T02:17:17.0799039Z     self.start(engine.DistMaster)
2021-10-04T02:17:17.0799772Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pytest_cov/plugin.py", line 222, in start
2021-10-04T02:17:17.0800329Z     self.cov_controller.start()
2021-10-04T02:17:17.0801082Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pytest_cov/engine.py", line 44, in ensure_topdir_wrapper
2021-10-04T02:17:17.0801675Z     return meth(self, *args, **kwargs)
2021-10-04T02:17:17.0802422Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/pytest_cov/engine.py", line 267, in start
2021-10-04T02:17:17.0803679Z     self.cov.start()
2021-10-04T02:17:17.0804569Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/control.py", line 555, in start
2021-10-04T02:17:17.0805141Z     self._inorout.warn_conflicting_settings()
2021-10-04T02:17:17.0805931Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/inorout.py", line 446, in warn_conflicting_settings
2021-10-04T02:17:17.0806803Z     self.warn("--include is ignored because --source is set", slug="include-ignored")
2021-10-04T02:17:17.0807773Z   File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/coverage/control.py", line 378, in _warn
2021-10-04T02:17:17.0808400Z     warnings.warn(msg, category=CoverageWarning, stacklevel=2)
2021-10-04T02:17:17.0809162Z coverage.exceptions.CoverageWarning: --include is ignored because --source is set (include-ignored)
2021-10-04T02:17:17.1133139Z ##[error]Bash exited with code '1'.
2021-10-04T02:17:17.1145767Z ##[section]Finishing: Run tests

```

--- a/.coveragerc
+++ b/.coveragerc
@@ -5,4 +5,4 @@ omit =
     scipy/setup.py
     scipy/*/setup.py
     scipy/signal/_max_len_seq_inner.py
-disable_warnings = --include is ignored because --source is set (include-ignored)
+disable_warnings = include-ignored

--- a/.coveragerc
+++ b/.coveragerc
@@ -5,3 +5,4 @@ omit =
     scipy/setup.py
     scipy/*/setup.py
     scipy/signal/_max_len_seq_inner.py
+disable_warnings = --include is ignored because --source is set (include-ignored)

